src/vm/tests/mock_tests/mock_test_allocate_with_re_enable_collection.rs

```diff
@@ -4,8 +4,9 @@ use crate::util::test_util::mock_method::*;
 use crate::util::test_util::mock_vm::*;
 use crate::AllocationSemantics;
 
-// This test allocates after calling initialize_collection(). When we exceed the heap limit, MMTk will trigger a GC. And block_for_gc will be called.
-// We havent implemented block_for_gc so it will panic. This test is similar to allocate_with_initialize_collection, except that we once disabled GC in the test.
+// This test allocates after calling initialize_collection(). When we exceed the heap limit for the first time, MMTk will not trigger GC since GC has been disabled
+// However, the second 1MB allocation will trigger a GC since GC is enabled again. And block_for_gc will be called.
+// We havent implemented block_for_gc so it will panic. This test is similar to allocate_with_initialize_collection, except that GC is disabled once in the test.
```

Comment:
This should be a doc comment (`///`, three slashes instead of two).  And there are also some minor Markdown formatting that we can conveniently fix now.

```suggestion
/// This test allocates after calling `initialize_collection()`. When we exceed the heap limit for the first time, MMTk will not trigger GC since GC has been disabled
/// However, the second 1MB allocation will trigger a GC since GC is enabled again. And `block_for_gc` will be called.
/// We haven't implemented `block_for_gc` so it will panic. This test is similar to `allocate_with_initialize_collection`, except that GC is disabled once in the test.
```

Reply:
About `block_for_gc` panicking:  I think we can re-attach another `MockMethod` after the first two `alloc` so that it actually expects `block_for_gc` will be called exactly once at the third `alloc`.